### PR TITLE
update opentelemetry reqs; unfreeze requests version

### PR DESCRIPTION
**Pull request by ruthenian8**
# Description

update opentelemetry reqs; unfreeze requests version

# Checklist

- [x] I have covered the code with tests
- [ ] I have added comments to my code to help others understand it
- [ ] I have updated the documentation to reflect the changes
- [x] I have performed a self-review of the changes

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -84,21 +84,22 @@ def merge_req_lists(*req_lists: List[str]) -> List[str]:
...
-    "opentelemetry-sdk==1.17.0",
-    "opentelemetry-semantic-conventions==0.38b0",
+    "opentelemetry-api>=1.20.0",
+    "opentelemetry-exporter-otlp>=1.20.0",
+    "opentelemetry-exporter-otlp-proto-grpc>=1.20.0",
+    "opentelemetry-exporter-otlp-proto-http>=1.20.0",
+    "opentelemetry-instrumentation>=0.38b0",
+    "opentelemetry-proto>=1.20.0",
+    "opentelemetry-sdk>=1.20.0",
+    "opentelemetry-semantic-conventions>=0.41b0",
```
Comment (RLKRo) [inline setup.py:94]: I think it'd be a good practice to add comments as to why there are restrictions.

File: setup.py [hunk 2]
```diff
@@ -84,21 +84,22 @@ def merge_req_lists(*req_lists: List[str]) -> List[str]:
...
 stats_dependencies = merge_req_lists(
     _sql_dependencies,
-    requests_requirements,
     otl_dependencies,
     [
+        "protobuf==3.20.3",
+        "requests",
         "wrapt==1.15.0",
         "tqdm==4.62.3",
         "omegaconf>=2.2.2",
```
Comment (RLKRo) [inline setup.py:105]: These are also extra dependencies and should be unrestricted (if possible).

**Review by RLKRo — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (RLKRo) [reply setup.py:94]: Most of the restrictions will be due to "latest api required".
This needs more context (like "required to export dictionaries").

File: setup.py [hunk 3]
```diff
@@ -84,24 +84,25 @@ def merge_req_lists(*req_lists: List[str]) -> List[str]:
...
 
 stats_dependencies = merge_req_lists(
     _sql_dependencies,
-    requests_requirements,
     otl_dependencies,
     [
-        "wrapt==1.15.0",
-        "tqdm==4.62.3",
-        "omegaconf>=2.2.2",
+        "protobuf==3.20.3",  # required by ydb & otlp
```
Comment (RLKRo) [inline setup.py:101]: I don't understand why this has to be frozen. Looking at the dep tree it seems like there are no conflicts.
```
> pipdeptree --reverse --packages protobuf
protobuf==3.20.3
├── googleapis-common-protos==1.60.0 [requires: protobuf>=3.19.5,<5.0.0.dev0,!=4.21.5,!=4.21.4,!=4.21.3,!=4.21.2,!=4.21.1,!=3.20.1,!=3.20.0]
│   ├── opentelemetry-exporter-otlp-proto-grpc==1.20.0 [requires: googleapis-common-protos~=1.52]
│   │   └── opentelemetry-exporter-otlp==1.20.0 [requires: opentelemetry-exporter-otlp-proto-grpc==1.20.0]
│   └── opentelemetry-exporter-otlp-proto-http==1.20.0 [requires: googleapis-common-protos~=1.52]
│       └── opentelemetry-exporter-otlp==1.20.0 [requires: opentelemetry-exporter-otlp-proto-http==1.20.0]
├── opentelemetry-proto==1.20.0 [requires: protobuf>=3.19,<5.0]
│   ├── opentelemetry-exporter-otlp-proto-common==1.20.0 [requires: opentelemetry-proto==1.20.0]
│   │   ├── opentelemetry-exporter-otlp-proto-grpc==1.20.0 [requires: opentelemetry-exporter-otlp-proto-common==1.20.0]
│   │   │   └── opentelemetry-exporter-otlp==1.20.0 [requires: opentelemetry-exporter-otlp-proto-grpc==1.20.0]
│   │   └── opentelemetry-exporter-otlp-proto-http==1.20.0 [requires: opentelemetry-exporter-otlp-proto-common==1.20.0]
│   │       └── opentelemetry-exporter-otlp==1.20.0 [requires: opentelemetry-exporter-otlp-proto-http==1.20.0]
│   ├── opentelemetry-exporter-otlp-proto-grpc==1.20.0 [requires: opentelemetry-proto==1.20.0]
│   │   └── opentelemetry-exporter-otlp==1.20.0 [requires: opentelemetry-exporter-otlp-proto-grpc==1.20.0]
│   └── opentelemetry-exporter-otlp-proto-http==1.20.0 [requires: opentelemetry-proto==1.20.0]
│       └── opentelemetry-exporter-otlp==1.20.0 [requires: opentelemetry-exporter-otlp-proto-http==1.20.0]
├── streamlit==1.26.0 [requires: protobuf>=3.20,<5]
│   └── streamlit-chat==0.1.1 [requires: streamlit>=0.63]
└── ydb==3.3.6 [requires: protobuf>=3.13.0,<5.0.0]
```

File: setup.py [hunk 4]
```diff
@@ -84,24 +84,25 @@ def merge_req_lists(*req_lists: List[str]) -> List[str]:
...
-    "opentelemetry-sdk==1.17.0",
-    "opentelemetry-semantic-conventions==0.38b0",
+    "opentelemetry-api>=1.20.0",  # latest api required
+    "opentelemetry-exporter-otlp>=1.20.0",  # latest api required
+    "opentelemetry-exporter-otlp-proto-grpc>=1.20.0",  # latest api required
+    "opentelemetry-exporter-otlp-proto-http>=1.20.0",  # latest api required
+    "opentelemetry-instrumentation>=0.38b0",
+    "opentelemetry-proto>=1.20.0",  # latest api required
+    "opentelemetry-sdk>=1.20.0",  # latest api required
+    "opentelemetry-semantic-conventions>=0.41b0",
```
Comment (RLKRo) [inline setup.py:94]: I also don't understand the need to specify all the packages. The package `opentelemetry-exporter-otlp` already has almost all the necessary packages:
```
opentelemetry-exporter-otlp==1.20.0
├── opentelemetry-exporter-otlp-proto-grpc [required: ==1.20.0, installed: 1.20.0]
│   ├── backoff [required: >=1.10.0,<3.0.0, installed: 2.2.1]
│   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│   │   └── wrapt [required: >=1.10,<2, installed: 1.15.0]
│   ├── googleapis-common-protos [required: ~=1.52, installed: 1.60.0]
│   │   └── protobuf [required: >=3.19.5,<5.0.0.dev0,!=4.21.5,!=4.21.4,!=4.21.3,!=4.21.2,!=4.21.1,!=3.20.1,!=3.20.0, installed: 3.20.3]
│   ├── grpcio [required: >=1.0.0,<2.0.0, installed: 1.57.0]
│   ├── opentelemetry-api [required: ~=1.15, installed: 1.20.0]
│   │   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│   │   │   └── wrapt [required: >=1.10,<2, installed: 1.15.0]
│   │   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.0.1]
│   │       └── zipp [required: >=0.5, installed: 3.16.2]
│   ├── opentelemetry-exporter-otlp-proto-common [required: ==1.20.0, installed: 1.20.0]
│   │   ├── backoff [required: >=1.10.0,<3.0.0, installed: 2.2.1]
│   │   └── opentelemetry-proto [required: ==1.20.0, installed: 1.20.0]
│   │       └── protobuf [required: >=3.19,<5.0, installed: 3.20.3]
│   ├── opentelemetry-proto [required: ==1.20.0, installed: 1.20.0]
│   │   └── protobuf [required: >=3.19,<5.0, installed: 3.20.3]
│   └── opentelemetry-sdk [required: ~=1.20.0, installed: 1.20.0]
│       ├── opentelemetry-api [required: ==1.20.0, installed: 1.20.0]
│       │   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│       │   │   └── wrapt [required: >=1.10,<2, installed: 1.15.0]
│       │   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.0.1]
│       │       └── zipp [required: >=0.5, installed: 3.16.2]
│       ├── opentelemetry-semantic-conventions [required: ==0.41b0, installed: 0.41b0]
│       └── typing-extensions [required: >=3.7.4, installed: 4.7.1]
└── opentelemetry-exporter-otlp-proto-http [required: ==1.20.0, installed: 1.20.0]
    ├── backoff [required: >=1.10.0,<3.0.0, installed: 2.2.1]
    ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
    │   └── wrapt [required: >=1.10,<2, installed: 1.15.0]
    ├── googleapis-common-protos [required: ~=1.52, installed: 1.60.0]
    │   └── protobuf [required: >=3.19.5,<5.0.0.dev0,!=4.21.5,!=4.21.4,!=4.21.3,!=4.21.2,!=4.21.1,!=3.20.1,!=3.20.0, installed: 3.20.3]
    ├── opentelemetry-api [required: ~=1.15, installed: 1.20.0]
    │   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
    │   │   └── wrapt [required: >=1.10,<2, installed: 1.15.0]
    │   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.0.1]
    │       └── zipp [required: >=0.5, installed: 3.16.2]
    ├── opentelemetry-exporter-otlp-proto-common [required: ==1.20.0, installed: 1.20.0]
    │   ├── backoff [required: >=1.10.0,<3.0.0, installed: 2.2.1]
    │   └── opentelemetry-proto [required: ==1.20.0, installed: 1.20.0]
    │       └── protobuf [required: >=3.19,<5.0, installed: 3.20.3]
    ├── opentelemetry-proto [required: ==1.20.0, installed: 1.20.0]
    │   └── protobuf [required: >=3.19,<5.0, installed: 3.20.3]
    ├── opentelemetry-sdk [required: ~=1.20.0, installed: 1.20.0]
    │   ├── opentelemetry-api [required: ==1.20.0, installed: 1.20.0]
    │   │   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
    │   │   │   └── wrapt [required: >=1.10,<2, installed: 1.15.0]
    │   │   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.0.1]
    │   │       └── zipp [required: >=0.5, installed: 3.16.2]
    │   ├── opentelemetry-semantic-conventions [required: ==0.41b0, installed: 0.41b0]
    │   └── typing-extensions [required: >=3.7.4, installed: 4.7.1]
    └── requests [required: ~=2.7, installed: 2.31.0]
        ├── certifi [required: >=2017.4.17, installed: 2023.7.22]
        ├── charset-normalizer [required: >=2,<4, installed: 3.2.0]
        ├── idna [required: >=2.5,<4, installed: 3.4]
        └── urllib3 [required: >=1.21.1,<3, installed: 2.0.4]
```
The only package from `otlp_dependencies` that will not be installed after running `pip install opentelemetry-exporter-otlp` is `opentelemetry-instrumentation`:

```
> pip install opentelemetry-exporter-otlp
> pip freeze
backoff==2.2.1
certifi==2023.7.22
charset-normalizer==3.2.0
Deprecated==1.2.14
googleapis-common-protos==1.60.0
grpcio==1.58.0
idna==3.4
importlib-metadata==6.8.0
opentelemetry-api==1.20.0
opentelemetry-exporter-otlp==1.20.0
opentelemetry-exporter-otlp-proto-common==1.20.0
opentelemetry-exporter-otlp-proto-grpc==1.20.0
opentelemetry-exporter-otlp-proto-http==1.20.0
opentelemetry-proto==1.20.0
opentelemetry-sdk==1.20.0
opentelemetry-semantic-conventions==0.41b0
protobuf==4.24.3
requests==2.31.0
typing_extensions==4.8.0
urllib3==2.0.5
wrapt==1.15.0
zipp==3.17.0
```

File: setup.py [hunk 5]
```diff
@@ -84,24 +84,25 @@ def merge_req_lists(*req_lists: List[str]) -> List[str]:
...
     otl_dependencies,
     [
-        "wrapt==1.15.0",
-        "tqdm==4.62.3",
-        "omegaconf>=2.2.2",
+        "protobuf==3.20.3",  # required by ydb & otlp
+        "requests",
+        "wrapt",
+        "tqdm",
+        "omegaconf>=2.2.2",  # latest api required
```
Comment (RLKRo) [inline setup.py:105]: Same, could you specify which aspects of the new version are needed.

**Review by RLKRo — CHANGES_REQUESTED** [review]

File: setup.py (hunk 4 above)
Comment (ruthenian8) [reply setup.py]: Updated

File: setup.py (hunk 5 above)
Comment (ruthenian8) [reply setup.py]: Updated

File: setup.py (hunk 3 above)
Comment (ruthenian8) [reply setup.py]: Updated

File: setup.py (hunk 1 above)
Comment (ruthenian8) [reply setup.py]: Updated

**Review by RLKRo — APPROVED** [review]

**Pull request merged**